cloudflare/queues/producer.go

```diff
@@ -0,0 +1,115 @@
+package queues
+
+import (
+	"errors"
+	"fmt"
+	"syscall/js"
+
+	"github.com/syumai/workers/cloudflare/internal/cfruntimecontext"
+	"github.com/syumai/workers/internal/jsutil"
+)
+
+type BatchMessage struct {
+	body    any
+	options *sendOptions
+}
+
+// NewBatchMessage creates a single message to be batched before sending to a queue.
+func NewBatchMessage(body any, opts ...SendOption) *BatchMessage {
+	options := defaultSendOptions()
+	for _, opt := range opts {
+		opt(options)
+	}
+	return &BatchMessage{body: body, options: options}
+}
+
+func (m *BatchMessage) toJS() (js.Value, error) {
+	if m == nil {
+		return js.Undefined(), errors.New("message is nil")
+	}
+
+	jsValue, err := m.options.ContentType.mapValue(m.body)
+	if err != nil {
+		return js.Undefined(), err
+	}
+
+	obj := jsutil.NewObject()
+	obj.Set("body", jsValue)
+	obj.Set("options", m.options.toJS())
+
+	return obj, nil
+}
+
+type Producer struct {
+	// queue - Objects that Queue API belongs to. Default is Global
+	queue js.Value
+}
+
+// NewProducer creates a new Producer object to send messages to a queue.
+// queueName is the name of the queue environment var to send messages to.
+// In Cloudflare API documentation, this object represents the Queue.
+//   - https://developers.cloudflare.com/queues/configuration/javascript-apis/#producer
+func NewProducer(queueName string) (*Producer, error) {
+	inst := cfruntimecontext.MustGetRuntimeContextEnv().Get(queueName)
+	if inst.IsUndefined() {
+		return nil, fmt.Errorf("%s is undefined", queueName)
+	}
+	return &Producer{queue: inst}, nil
+}
+
+// Send sends a single message to a queue. This function allows setting send options for the message.
+// If no options are provided, the default options are used (QueueContentTypeJSON and no delay).
+//
+//   - https://developers.cloudflare.com/queues/configuration/javascript-apis/#producer
+//   - https://developers.cloudflare.com/queues/configuration/javascript-apis/#queuesendoptions
+func (p *Producer) Send(content any, opts ...SendOption) error {
```

Comment:
In Go, `Send` method with `any` content type doesn't seem useful enough.
I think `ContentType` send option should be removed and separated `Send` methods should be declared.

e.g.
```go
func (p *Producer) SendText(content string, opts ...SendOption) error {}
func (p *Producer) SendBytes(content []byte, opts ...SendOption) error {}
func (p *Producer) SendJSON(content any, opts ...SendOption) error {}
func (p *Producer) SendV8(content js.Value, opts ...SendOption) error {}
```

I'll separate current implementation after.